component/src/main/java/org/wso2/carbon/identity/authenticator/office365/Office365Authenticator.java

```diff
@@ -159,8 +159,24 @@ public String getContextIdentifier(HttpServletRequest request) {
      * Check whether the authentication or logout request can be handled by the authenticator
      */
     @Override
-    public boolean canHandle(HttpServletRequest arg0) {
-        return true;
+    public boolean canHandle(HttpServletRequest request) {
+
+        if (request.getParameter(Office365AuthenticatorConstants.CODE) != null &&
+                request.getParameter(OIDCAuthenticatorConstants.OAUTH2_PARAM_STATE) != null &&
+                Office365AuthenticatorConstants.AUTHENTICATOR_FRIENDLY_NAME.equals(getLoginType(request))) {
+            return true;
+        }
+
+        return false;
+    }
+
+    protected String getLoginType(HttpServletRequest request) {
+        String state = request.getParameter(OIDCAuthenticatorConstants.OAUTH2_PARAM_STATE);
+        if (state != null) {
+            return state.split(",")[1];
```

Comment:
If there is no "," character in the state, this can lead to a null pointer. This is there in other authenticators too. But it's better to first fix here.